pkg/commands/git_commands/remote_loader.go

```diff
@@ -83,7 +84,20 @@ func (self *RemoteLoader) GetRemotes() ([]*models.Remote, error) {
 func (self *RemoteLoader) getRemoteBranchesByRemoteName() (map[string][]*models.RemoteBranch, error) {
 	remoteBranchesByRemoteName := make(map[string][]*models.RemoteBranch)
 
-	cmdArgs := NewGitCmd("branch").Arg("-r").ToArgv()
+	sortOrder := "refname"
+	switch self.UserConfig.Git.RemoteBranchOrder {
+	case "alphabetical":
+		sortOrder = "refname"
+	case "lastCommit":
+		sortOrder = "-authordate"
+	}
+
+	cmdArgs := NewGitCmd("for-each-ref").
+		Arg(fmt.Sprintf("--sort=%s", sortOrder)).
+		Arg("--format=%(refname:short)").
+		Arg("refs/remotes").
+		ToArgv()
+
 	err := self.cmd.New(cmdArgs).DontLog().RunAndProcessLines(func(line string) (bool, error) {
 		// excluding lines like 'origin/HEAD -> origin/master' (there will be a separate
```

Comment:
This is probably no longer necessary as `for-each-ref refs/remotes` doesn't output the HEAD -> branch  lines.